### PR TITLE
Added changes for issue #15 and #17

### PR DESCRIPTION
**Pull request by Himan10**
Changes implemented for the issue number #15 

Please don't merge this PR because the seed_files we have on this repo. aren't updated, will update and then merge.

### CODE REVIEW

File: apps/accounts/views.py [hunk 1]
```diff
@@ -115,6 +117,19 @@ def post(self, request, format=None):
...
         token = OTP_DummyToken(user,"verify")
+
+        # Add an entry in the tbl_user_profile with dummy data
+        dummy_data = {
+            "user_id": user.user_id.hex,
+            "name" : user.name,
+            "email": user.email,
+            "user_type": user.user_type,
+            "about": "N/A"
+        }
```
Comment (HimanshuSharma-Angel) [inline apps/accounts/views.py:128]: can we about as null

**Review by HimanshuSharma-Angel — APPROVED** [review]

**Comment by Himan10** [comment]
Latest commit contains the changes for issue number #17 
Sorry forgot to create another branch :(

**Comment by Himan10** [comment]
Also, make sure another person from the team (either anubhav or parth) perform the testing of this PR, if everything works fine, then you can merge.

I have also tested everything from my end, and on my system, it's working perfectly but as of now I have tested this on local db.

File: apps/jobs/views.py [hunk 2]
```diff
@@ -289,11 +297,63 @@ class UserViewSets(viewsets.ModelViewSet):
     queryset = User.objects.all()
     serializer_class = UserSerializer
 
-    def create(self, request, *args, **kwargs):
+    def update(self, request, *args, **kwargs):
         """
         Overriding the create method (used in POST request),
         This method creates a new user profile in the database.
         """
+        common_response_parameters = {
```
Comment (hims1911) [inline apps/jobs/views.py:305]: put in constant package

File: apps/jobs/views.py [hunk 3]
```diff
@@ -289,11 +297,63 @@ class UserViewSets(viewsets.ModelViewSet):
...
+                return Response(
+                    {"error": "AccessToken is not valid"}, **common_response_parameters
+                )
+            except Exception as err:
+                common_response_parameters[
+                    "status"
+                ] = status.HTTP_500_INTERNAL_SERVER_ERROR
+                return Response(
+                    {"error": "Something went wrong"}, **common_response_parameters
+                )
```
Comment (hims1911) [inline apps/jobs/views.py:326]: keep different response under constant/response.py

File: apps/jobs/views.py [hunk 4]
```diff
@@ -318,16 +378,70 @@ def create(self, request, *args, **kwargs):
...
-        headers = self.get_success_headers(data)
-        return Response(
-            serializer.data, status=status.HTTP_201_CREATED, headers=headers
-        )
+        # Check if the given user_id actually belongs to the user
+        # This check is necessary because only specific users can update their profile
+
+        # get the user_id from the url path
+        if user_id := search("\w{32}", request.stream.path):
+            user_id = user_id[0]
```
Comment (hims1911) [inline apps/jobs/views.py:386]: please follow security checks

File: apps/jobs/views.py [hunk 5]
```diff
@@ -318,16 +378,70 @@ def create(self, request, *args, **kwargs):
...
+
+        # if case, if someone crafts a request in such a way that
+        # user_id in accessToken and user_id in the api path are same
+
+        # perform check on database as well
+        if user_id == payload["user_id"]:
```
Comment (hims1911) [inline apps/jobs/views.py:392]: we dont have to check, fetch it from payload itself

**Review by hims1911 — APPROVED** [review]

File: apps/accounts/views.py (hunk 1 above)
Comment (Himan10) [reply apps/accounts/views.py:128]: Replaced the hardcoded value "N/A" with `None`

File: apps/jobs/views.py (hunk 2 above)
Comment (Himan10) [reply apps/jobs/views.py]: Moved this dictionary and other necessary response related items in `constants/response.py`

File: apps/jobs/views.py (hunk 3 above)
Comment (Himan10) [reply apps/jobs/views.py]: Created a function in `constants/response.py` where you need to pass only the message and status code, and it'll return the Response object with message, status and content_type

File: apps/jobs/views.py (hunk 4 above)
Comment (Himan10) [reply apps/jobs/views.py]: Removed the check for user_id comparison, now only the user_id present in access-token will be used in the update method, especially for updating user_information

File: apps/jobs/views.py (hunk 5 above)
Comment (Himan10) [reply apps/jobs/views.py]: Removed this comparison.

**Comment by Himan10** [comment]
Till now I have added only few constants for the Jobs app, once I'll start anubhav's account app code-cleaning, will update "accounts" side of constants in this as well.

**Pull request merged**